0278-template-only-components.md

```diff
@@ -0,0 +1,249 @@
+
+However, this API does not work for template-only components, as they do
+not have a component JavaScript class by definition. Therefore, in practice,
+template-only components always come with a `<div>` wrapper, along with the
+default `id` and `class` attributes, with no obvious ways to customize it.
+
+This is quite problematic, as it is often desirable to use a template-only
+component to organize content that requires a certain markup structure. The
+most common workaround for this problem is to use a partial instead, which
+comes with [a host of issues](https://github.com/emberjs/rfcs/pull/262). I
+will discuss other workarounds in the section below.
+
+This RFC proposes to add a global flag to remove this wrapper element around
+template-only components. This will allow the component author to specify the
+wrapper element in the component template, offering direct control over the
+tag name and other attributes. It would also allow the component to have more
+than one top-level element, or none at all.
+
+In other words, this flag changes template-only components in the app to have
+"Outer HTML" semantics. _What you type is what you get._
+
+Notably, [Glimmer Components](https://glimmerjs.com/guides/templates-and-helpers)
+have adopted the "Outer HTML" semantics long ago and the experience has been
+very positive. This would be one of the first pieces of the Glimmer.js experiment
+to make its way into Ember. We think this feature is small, self-contained but
+useful enough to be integrated back into Ember at this point.
+
+If accepted, this RFC will fully subsume the [Non-context-shifting partials](https://github.com/emberjs/rfcs/pull/262)
+RFC. We can therefore (at a later time, in a separate RFC) explore deprecating
+partials in favor of wrapper-free template-only components.
+
+# Detailed design
+
+## API Surface
+
+We should not expose the flag directly as a public API. Instead, we should
+abstract the flag with a "privileged addon" whose only purpose is to enable
+the flag. Applications will enable the flag by installing this addon. This
+will allow for more flexibility in changing the flag's implementation (the
+location, naming, value, or even the existence of it) in the future. From the
+user's perspective, it is the _addon_ that provides thisfunctionality. The
+flag is simply an internal implementation detail.
+
+We have done this before in other cases (such as the legacy view addon during
+the 2.0 transition period), and it has generally worked well.
+
+When landing this feature, it will be entirely opt-in for existing apps, but
+the Ember CLI application blueprint should be updated to include the addon by
+default. At a later time, we should provide another addon that _disables_ the
+flag explicitly (installing both addons would be an install-time error). At
+that time, we will issue a deprecation warning if the flag is *not set*, with
+a message that directs the user to install one of the two addons.
```

Comment:
IMHO the end goal should be that the default setting for the flag is "enabled" (aka. no wrapper), and that we don't need to include the enable-addon anymore. Is that the plan?

Reply:
Yes. I think that should be the case in Ember 4.0.